t/path.t

```diff
@@ -67,3 +67,38 @@ metadata multipe path 1
 metadata multipe path 2
 nil
 nil
+
+
+=== TEST 3: multiple path with overlap
+--- config
+    location /t {
+        content_by_lua_block {
+            local radix = require("resty.radixtree")
+            local rx = radix.new({
+                {
+                    paths = {"/equal*"},
+                    metadata = "metadata multipe path 1",
+                },
+                {
+                    paths = {"/equal123*"},
+                    metadata = "metadata multipe path 2",
+                },
+            })
+
+            ngx.say(rx:match("/equal"))
```

Comment:
I run the test case at my local machine right now, the test cases are always passed whether or not the `src/rax.c` was patched.

The test case can catch this bug when I remove `ngx.say(rx:match("/equal"))`.